pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -91,21 +111,47 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 		return ssn.PredicateFn(task, node)
 	}
 
+	// To pick <namespace, queue> tuple for job, we choose to pick namespace firstly.
+	// Because we believe that number of queues would less than namespaces in most case.
+	// And, this action would make the resource usage among namespace balanced.
 	for {
-		if queues.Empty() {
+		if namespaces.Empty() {
 			break
 		}
 
-		queue := queues.Pop().(*api.QueueInfo)
-		if ssn.Overused(queue) {
-			glog.V(3).Infof("Queue <%s> is overused, ignore it.", queue.Name)
-			continue
+		// pick namespace from namespaces PriorityQueue
+		namespace := namespaces.Pop().(api.NamespaceName)
+
+		queueInNamespace := jobsMap[namespace]
```

Comment:
A little confused. What is the relationship between queue and namespace? According to the doc [motivation](https://github.com/volcano-sh/volcano/blob/master/docs/design/fairshare.md#motivation), a namespace works like a user, and there might be many users(namespaces) in a queue. Will be many queues in a namespace?

Reply:
@hex108 As I understand,  generally speaking, different users (namespace) could schedule jobs in different Queues.  This doc I think concentrates more on the faire resource sharing in one queue between users. This solution is mainly trying to resolve starvation problem in resource sharing.